### ISSUE TITLE
FOQUS Cloud Dropped Job

### ISSUE THREAD

**Issue by boverhof** [CCSI-Toolset/FOQUS#687]
# Description
After running optimization for a while a job errors and the next job in the batch is. dropped. and never appears to make it in DynamoDB table.

## Dropped Job 
- Id: e054c952-013d-4609-b213-b3d9a09fb08c
- SessionId: 747919db-bf9c-4e83-b4ef-b0b033ad985c
```
2020-03-05T04:17:34.730Z post-session-start Payload: 
{
    "Simulation": "zzfoqus_BFB",
    "Input": {
        "graph": {},
        "BFB": {
            "BFBadsB.Dt": 11.89286292001983,
            "BFBadsB.dx": 0.0127,
            "BFBadsB.Lb": 2.0851517017356476,
            "BFBadsM.Dt": 15,
            "BFBadsM.dx": 0.06685903722891832,
            "BFBadsM.Lb": 1.9739825608135484,
            "BFBadsT.Dt": 15,
            "BFBadsT.dx": 0.05290146940394774,
            "BFBadsT.Lb": 2.204472530448875,
            "BFBRGN.Dt": 9.027874772897324,
            "BFBRGN.Lb": 8.888762230394914,
            "BFBRGNTop.Dt": 9.199673642915357,
            "BFBRGNTop.Lb": 7.216528421796692,
            "GHXfg.A_exch": 16371.202308779422,
            "Kd": 100,
            "BFBadsB.Cr": 1,
            "BFBadsM.Cr": 1,
            "BFBadsT.Cr": 1,
            "BFBRGN.Cr": 1,
            "BFBRGNTop.Cr": 1,
            "dp": 0.0001598,
            "GHXfg.GasIn.P": 1.01325,
            "GHXfg.GasIn.T": 54,
            "fg_flow": 70263.9
        }
    },
    "Reset": false,
    "Id": "e054c952-013d-4609-b213-b3d9a09fb08c",
    "resource": "job",
    "status": "submit",
    "jobid": "e054c952-013d-4609-b213-b3d9a09fb08c",
    "sessionid": "747919db-bf9c-4e83-b4ef-b0b033ad985c",
    "event": "status"
}
```

### post-session-start cloudwatch log
```
04:17:34 START RequestId: b645a5ee-bc6d-4172-906f-edbcc3165163 Version: $LATEST
04:17:34.352Z post-session-start Running index.handler: "POST"
04:17:34.368Z post-session-start request: {"resource":"/session/{id}/start","path":"/session/747919db-bf9c-4e83-b4ef-b0b033ad985c/start","httpMethod":"POST","headers":{"Accept-Encoding":"identity","Authorization":"Basic Ynlyb246c29lcHlhbg==","Content-Type":"application/json; charset=utf-8","Host":"b7x9ucxadg.execute-api.us-east-1.amazonaws.com","User-Agent":"Python-urllib/3.7","X-Amzn-T
04:17:34.387Z post-session-start PATH: /session/747919db-bf9c-4e83-b4ef-b0b033ad985c/start
04:17:34.388Z post-session-start SESSIONID: 747919db-bf9c-4e83-b4ef-b0b033ad985c
04:17:34.388Z post-session-start SESSION BUCKET_NAME: foqussessiondev1563311335
T04:17:34.390Z post-session-start ==================================
04:17:34.390Z post-session-start Stopping index.handler
04:17:34.497Z post-session-start SUCCESS: {"IsTruncated":false,"Contents":[{"Key":"byron/session/create/747919db-bf9c-4e83-b4ef-b0b033ad985c/1583381853714.json","LastModified":"2020-03-05T04:17:34.000Z","ETag":"\"ee10f3876cd328ecb0c26e37698cf06f\"","Size":3402,"StorageClass":"STANDARD"}],"Name":"foqussessiondev1563311335","Prefix":"byron/session/create/747919db-bf9c-4e83-b4ef-b0b033ad98
04:17:34.497Z post-session-start LEN: 1
04:17:34.497Z post-session-start Create Topic: Name=FOQUS-Gateway-FOQUSUpdateTopic-19K4KKZTEJNJL
04:17:34.568Z post-session-start SUCCESS: {"ResponseMetadata":{"RequestId":"679ece8f-1d80-542e-a8ca-9855621ec14a"},"TopicArn":"arn:aws:sns:us-east-1:387057575688:FOQUS-Gateway-FOQUSUpdateTopic-19K4KKZTEJNJL"}
04:17:34.568Z post-session-start PARAMS: {"Bucket":"foqussessiondev1563311335","Key":"byron/session/create/747919db-bf9c-4e83-b4ef-b0b033ad985c/1583381853714.json"}
04:17:34.672Z post-session-start SESSION(747919db-bf9c-4e83-b4ef-b0b033ad985c): Notify Starting 5
04:17:34.672Z post-session-start INDEX: 0
04:17:34.672Z post-session-start Payload: {"Simulation":"zzfoqus_BFB","Input":{"graph":{},"BFB":{"BFBadsB.Dt":11.89286292001983,"BFBadsB.dx":0.0127,"BFBadsB.Lb":2.0851517017356476,"BFBadsM.Dt":15,"BFBadsM.dx":0.06685903722891832,"BFBadsM.Lb":1.9739825608135484,"BFBadsT.Dt":15,"BFBadsT.dx":0.05290146940394774,"BFBadsT.Lb":2.204472530448875,"BFBRGN.Dt":9.027874772897324,"BFBRGN.Lb":8.8887
04:17:34.674Z post-session-start INDEX: 1
04:17:34.674Z post-session-start Payload: {"Simulation":"zzfoqus_BFB","Input":{"graph":{},"BFB":{"BFBadsB.Dt":11.89286292001983,"BFBadsB.dx":0.0127,"BFBadsB.Lb":2.0851517017356476,"BFBadsM.Dt":15,"BFBadsM.dx":0.06685903722891832,"BFBadsM.Lb":1.9739825608135484,"BFBadsT.Dt":15,"BFBadsT.dx":0.05290146940394774,"BFBadsT.Lb":2.204472530448875,"BFBRGN.Dt":9.027874772897324,"BFBRGN.Lb":8.8887
04:17:34.727Z post-session-start INDEX: 2
04:17:34.728Z post-session-start Payload: {"Simulation":"zzfoqus_BFB","Input":{"graph":{},"BFB":{"BFBadsB.Dt":11.89286292001983,"BFBadsB.dx":0.0127,"BFBadsB.Lb":2.0851517017356476,"BFBadsM.Dt":15,"BFBadsM.dx":0.06685903722891832,"BFBadsM.Lb":1.9739825608135484,"BFBadsT.Dt":15,"BFBadsT.dx":0.05290146940394774,"BFBadsT.Lb":2.204472530448875,"BFBRGN.Dt":9.027874772897324,"BFBRGN.Lb":8.8887
04:17:34.730Z post-session-start INDEX: 3
04:17:34.730Z post-session-start Payload: {"Simulation":"zzfoqus_BFB","Input":{"graph":{},"BFB":{"BFBadsB.Dt":11.89286292001983,"BFBadsB.dx":0.0127,"BFBadsB.Lb":2.0851517017356476,"BFBadsM.Dt":15,"BFBadsM.dx":0.06685903722891832,"BFBadsM.Lb":1.9739825608135484,"BFBadsT.Dt":15,"BFBadsT.dx":0.05290146940394774,"BFBadsT.Lb":2.204472530448875,"BFBRGN.Dt":9.027874772897324,"BFBRGN.Lb":8.8887
04:17:34.788Z post-session-start INDEX: 4
04:17:34.788Z post-session-start Payload: {"Simulation":"zzfoqus_BFB","Input":{"graph":{},"BFB":{"BFBadsB.Dt":11.89286292001983,"BFBadsB.dx":0.0127,"BFBadsB.Lb":2.0851517017356476,"BFBadsM.Dt":15,"BFBadsM.dx":0.06685903722891832,"BFBadsM.Lb":1.9739825608135484,"BFBadsT.Dt":15,"BFBadsT.dx":0.05290146940394774,"BFBadsT.Lb":2.204472530448875,"BFBRGN.Dt":9.027874772897324,"BFBRGN.Lb":8.8887
04:17:34.789Z post-session-start Delete : [object Object]
04:17:35.088Z post-session-start handleDelete: DELETED undefined
04:17:35
END RequestId: b645a5ee-bc6d-4172-906f-edbcc3165163
```

**Comment by boverhof** [comment]
# Issue 1:  ValidationError on submit job in foqus-sns-update 
- no consumerId on submit job, need to add it only if available
```

Error: Error: "object"
Error: Error: "ValidationException: Value provided in ExpressionAttributeNames unused in expressions: keys: {#c}
    at Request.extractError (/Users/boverhof/Documents/workspace.ccsi2/FOQUS/cloud/aws/node/foqus-sns-update/node_modules/aws-sdk/lib/protocol/json.js:51:27)
    at Request.callListeners (/Users/boverhof/Documents/workspace.ccsi2/FOQUS/cloud/aws/node/foqus-sns-update/node_modules/aws-sdk/lib/sequential_executor.js:106:20)
    at Request.emit (/Users/boverhof/Documents/workspace.ccsi2/FOQUS/cloud/aws/node/foqus-sns-update/node_modules/aws-sdk/lib/sequential_executor.js:78:10)
    at Request.emit (/Users/boverhof/Documents/workspace.ccsi2/FOQUS/cloud/aws/node/foqus-sns-update/node_modules/aws-sdk/lib/request.js:683:14)
    at Request.transition (/Users/boverhof/Documents/workspace.ccsi2/FOQUS/cloud/aws/node/foqus-sns-update/node_modules/aws-sdk/lib/request.js:22:10)
    at AcceptorStateMachine.runTo (/Users/boverhof/Documents/workspace.ccsi2/FOQUS/cloud/aws/node/foqus-sns-update/node_modules/aws-sdk/lib/state_machine.js:14:12)
    at /Users/boverhof/Documents/workspace.ccsi2/FOQUS/cloud/aws/node/foqus-sns-update/node_modules/aws-sdk/lib/state_machine.js:26:10
    at Request.<anonymous> (/Users/boverhof/Documents/workspace.ccsi2/FOQUS/cloud/aws/node/foqus-sns-update/node_modules/aws-sdk/lib/request.js:38:9)
    at Request.<anonymous> (/Users/boverhof/Documents/workspace.ccsi2/FOQUS/cloud/aws/node/foqus-sns-update/node_modules/aws-sdk/lib/request.js:685:12)
    at Request.callListeners (/Users/boverhof/Documents/workspace.ccsi2/FOQUS/cloud/aws/node/foqus-sns-update/node_modules/aws-sdk/lib/sequential_executor.js:116:18)"
(py3) Joshuas-MacBook-Pro:foqus-sns-update boverhof$ 
```

**Comment by boverhof** [comment]
# CloudWatch post-session-append 
- 5 items went in DynamoDB but only see 4.
- TTL is set low for `create` jobs, and think because `submit`, `setup` and `running` updates failed on ValidateError it never got to `output` and `success` and therefore TTL expired and it was removed.
```
START RequestId: 8787e063-199a-4f3e-89e4-5c05ea886071 Version: $LATEST
2020-03-05T04:17:33.713Z post-session-append Running index.handler: "POST"
2020-03-05T04:17:33.714Z post-session-append parseBodyJSON
2020-03-05T04:17:33.715Z post-session-append writeToDynamo:  count=5
2020-03-05T04:17:33.715Z post-session-append Dynamodb.batchWrite to FOQUS_Resources
2020-03-05T04:17:33.870Z post-session-append checkForUnprocessedItems: 
{
    "UnprocessedItems": {}
}

2020-03-05T04:17:33.870Z post-session-append Zero UnprocessedItems
2020-03-05T04:17:33.870Z post-session-append putS3 items: 5
2020-03-05T04:17:33.871Z post-session-append putS3(foqussessiondev1563311335):  byron/session/create/747919db-bf9c-4e83-b4ef-b0b033ad985c/1583381853714.json
2020-03-05T04:17:33.983Z post-session-append handleDone
END RequestId: 8787e063-199a-4f3e-89e4-5c05ea886071
```

**Comment by ksbeattie** [comment]
@boverhof is this one still an issue?  Was it closed by #689?
